package-builder/flox-build.mk

```diff
@@ -195,23 +206,45 @@ define CLEAN_result_link_template =
     $(if $(_storePath),clean_result_storepath/$(1))
 endef
 
+# The following env vars need to be passed from the outer "develop" environment
+# to the inner "wrapper" environment in the in-situ build mode in support of
+# the tools and compilers found in the outer environment, and for Flox to
+# otherwise function properly.
+ALLOW_OUTER_ENV_VARS := FLOX_RUNTIME_DIR HOME PATH \
+  $(filter NIX_CFLAGS% NIX_CC%,$(.VARIABLES))
```

Comment:
for my understanding, this "retains" `NIX_CFLAGS` and `NIX_CC`, and drops everything else, right?
While NIX_CC and NIX_CFLAGS might be enough in the case we ran into, there are a couple more `NIX_` vars that are relevant to the cc wrappers, and may be relevant here:


```
NIX_HARDENING_ENABLE                                   
NIX_IGNORE_LD_THROUGH_GCC                              
NIX_LDFLAGS                                            
NIX_BINTOOLS                                                                                                                                    
NIX_BUILD_CORES
NIX_NO_SELF_RPATH
NIX_CC
NIX_CFLAGS_COMPILE                                                                             
NIX_COREFOUNDATION_RPATH                                 
NIX_DONT_SET_RPATH                                                                    
NIX_ENFORCE_NO_NATIVE                                         


# not sure about these, they look cross comp related
NIX_BINTOOLS_WRAPPER_TARGET_HOST_aarch64_apple_darwin
NIX_CC_WRAPPER_TARGET_HOST_aarch64_apple_darwin
NIX_PKG_CONFIG_WRAPPER_TARGET_HOST_aarch64_apple_darwin
NIX_DONT_SET_RPATH_FOR_BUILD
NIX_DONT_SET_RPATH_FOR_TARGET                                                                        
```

Reply:
The `%` in the `filter` command is a wildcard search, so it will retain `NIX_CFLAGS*` and `NIX_CC*` (in a glob sense). But you're right that all those other variables will not be preserved, and perhaps they should. In truth I came up with that list empirically through experimentation, so we may find we need to add others over time, or abandon this approach altogether and come up with another way. But what I can say is that these are the _minimum_ variables required to get our tests to pass so that's why I chose them.

Reply:
Going to resolve this for now, but we should keep an eye on this list, I think that may end up causing hard to inconsistencies.